### ISSUE TITLE
Run functional tests from make check

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#18816]
We have a bunch of tests:

* unit tests
* util tests written in python
* bench runner
* subtree tests

All of them are run when you type `make check`.

However, for running the functional tests, one has to type `./test/functional/test_runner.py`

I think some people asked for the functional tests to be run as part of `make check`, or maybe a different target like `make check-functional`?

**Comment by mrwhythat** [comment]
Perhaps both add a new `check-functional` target that does `./test/functional/test_runner.py` and make it the final step of `check` target?

**Comment by practicalswift** [comment]
Personally I like that `make check` is relatively quick thanks to not running the heavy functional tests. 

What about adding another target (say `make check-all`?) which runs `make check` + `test/functional/test_runner.py`?

**Comment by MarcoFalke** [comment]
The unit tests run in 1 minute and 9 seconds on my system. Mostly because the libsecp tests are not run in parallel. The functional tests run in under 3 minutes.

An alternative would be to have `./configure --with-functional-tests --without-subtree-tests && make check` run Bitcoin Core tests only.

**Comment by brakmic** [comment]
Hi,

I have played a bit with Makefiles and came with this cheap "solution". Not sure if idiomatic or proper in any way, but it works.

* Add new `Makefile` file to top *test* directory with this content:

```shell
check-functional:
  $(PYTHON) ../test/functional/test_runner.py
```

* Change Makefile.am

```diff
diff --git a/Makefile.am b/Makefile.am
index c35f5080a..bcf3b064c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -342,3 +342,6 @@ clean-local: clean-docs
        rm -rf coverage_percent.txt test_bitcoin.coverage/ total.coverage/ fuzz.coverage/ test/tmp/ cache/ $(OSX_APP)
        rm -rf test/functional/__pycache__ test/functional/test_framework/__pycache__ test/cache share/rpcauth/__pycache__
        rm -rf osx_volname dist/ dpi36.background.tiff dpi72.background.tiff
+
+check-functional:
+       $(MAKE) -C $(top_builddir)/test
```

* test run with `make check-functional`

![make_check_functional](https://raw.githubusercontent.com/brakmic/bazaar/master/images/random/make_check_functional_2.png)

**Comment by hebasto** [comment]
I agree to have a `make` target to run all of the tests. But I'd preserve a target which runs current set of tests without functional ones.

**Comment by jonatack** [comment]
I use a bash alias to run the unit tests followed by the functional tests. Agree with keeping the current `make check` default and maybe adding an additional `make check -all` or similar.

**Comment by AmitNudel** [comment]
I'm looking to contribute for the first time to open source. Is it still an open issue? I'd like to try and help.

**Comment by ismaelsadeeq** [comment]
Hello! I noticed this issue and would like to contribute by enabling running the tests in ./test/functional/test_runner.py with `make check-functional`. My plan is to also add a `make check-all` target to run both `make check` and `make check-functional` tests.
Any suggestions on how to make the solution even better, I'm all ears! Thank you so much for your time and help.
